File: benchmarks/bench_filter_map.py
import time

nums = list(range(1, 10_000_001))

start = time.perf_counter()
result = [n * n for n in nums if n % 2 == 0]
elapsed = (time.perf_counter() - start) * 1000

print(f"{elapsed:.4f}")
print(f"len = {len(result)}", file=__import__('sys').stderr)

File: benchmarks/bench_filter_map.rs
use std::time::Instant;

fn main() {
    let nums: Vec<i32> = (1..=10_000_000).collect();

    let start = Instant::now();
    let result: Vec<i64> = nums.iter()
        .filter(|&&n| n % 2 == 0)
        .map(|&n| (n as i64) * (n as i64))
        .collect();
    let elapsed = start.elapsed();

    println!("{}", elapsed.as_secs_f64() * 1000.0);
    eprintln!("len = {}", result.len());
}
